crates/paimon/src/spec/types.rs

```diff
@@ -1338,4 +1496,204 @@ mod tests {
+
+        // DateType
+        let date_type: DateType = DateType::with_nullable(true);
+        let serialized: String = serde_json::to_string(&date_type).unwrap();
+        assert_eq!(serialized, json!("DATE").to_string());
+        let date_type: DateType = DateType::with_nullable(false);
+        let serialized: String = serde_json::to_string(&date_type).unwrap();
+        assert_eq!(serialized, json!("DATE NOT NULL").to_string());
+
+        // LocalZonedTimestampType
+        let localzoned_type: LocalZonedTimestampType =
+            LocalZonedTimestampType::with_nullable(true, 6).unwrap();
+        let serialized: String = serde_json::to_string(&localzoned_type).unwrap();
+        assert_eq!(
+            serialized,
+            json!("TIMESTAMP WITH LOCAL TIME ZONE(6)").to_string()
+        );
+        let localzoned_type: LocalZonedTimestampType =
+            LocalZonedTimestampType::with_nullable(false, 6).unwrap();
+        let serialized: String = serde_json::to_string(&localzoned_type).unwrap();
+        assert_eq!(
+            serialized,
+            json!("TIMESTAMP WITH LOCAL TIME ZONE(6) NOT NULL").to_string()
+        );
+
+        // TimeType
+        let time_type: TimeType = TimeType::with_nullable(true, 6).unwrap();
+        let serialized: String = serde_json::to_string(&time_type).unwrap();
+        assert_eq!(serialized, json!("TIME(6)").to_string());
+        let time_type: TimeType = TimeType::with_nullable(false, 6).unwrap();
+        let serialized: String = serde_json::to_string(&time_type).unwrap();
+        assert_eq!(serialized, json!("TIME(6) NOT NULL").to_string());
+        let time_type: TimeType = TimeType::with_nullable(false, 6).unwrap();
+        let serialized: String = serde_json::to_string(&time_type).unwrap();
+        assert_eq!(serialized, json!("TIME(6) NOT NULL").to_string());
+
+        // TimestampType
+        let timestamp_type: TimestampType = TimestampType::with_nullable(true, 6).unwrap();
+        let serialized: String = serde_json::to_string(&timestamp_type).unwrap();
+        assert_eq!(serialized, json!("TIMESTAMP(6)").to_string());
+        let timestamp_type: TimestampType = TimestampType::with_nullable(false, 6).unwrap();
+        let serialized: String = serde_json::to_string(&timestamp_type).unwrap();
+        assert_eq!(serialized, json!("TIMESTAMP(6) NOT NULL").to_string());
+
+        // ArrayType
+        let int_type: DataType = DataType::Int(IntType::with_nullable(true));
+        let array_type: ArrayType = ArrayType::with_nullable(true, int_type.clone());
+        let serialized: String = serde_json::to_string(&array_type).unwrap();
+        assert_eq!(serialized, json!("ARRAY<INTEGER>").to_string());
+        let array_type: ArrayType = ArrayType::with_nullable(false, int_type.clone());
+        let serialized: String = serde_json::to_string(&array_type).unwrap();
+        assert_eq!(serialized, json!("ARRAY<INTEGER> NOT NULL").to_string());
```

Comment:
This serialized json is not right. These four types have different representation from the `SerializeDisplay`

![image](https://github.com/user-attachments/assets/88e0462c-9a8f-41e3-b72a-4eb90895a41e)

Reply:
<img width="364" alt="image" src="https://github.com/user-attachments/assets/3ff73fa3-ad96-4fe3-911d-364c5863a90d">

The serialized version for `ARRAY<STRING>`

Reply:
I'm a bit confused. @JingsongLi mentioned that Paimon will use `as_sql_string` everywhere. I'll double-check this part.